stl/inc/flat_set

```diff
@@ -571,29 +631,35 @@ private:
             return;
         }
 
-        _Sort_potentially_sorted(_Begin, _End);
-        _STL_INTERNAL_CHECK(_STD is_sorted(begin(), end(), _Get_comp()));
+        // O(N) if already sorted.
+        const key_compare& _Compare    = _Get_comp();
+        const iterator _Begin_unsorted = _STD is_sorted_until(_Begin, _End, _Compare);
+
+        _STD sort(_Begin_unsorted, _End, _Compare);
+        _STD inplace_merge(begin(), _Begin_unsorted, _End, _Compare);
+
+        _STL_INTERNAL_CHECK(_STD is_sorted(_Begin, _End, _Compare));

```

Comment:
After the code fusion here, we can pass `_Begin` instead of `begin()` to `inplace_merge()`.